x-pack/plugins/apm/public/components/alerting/alerting_flyout/index.tsx

```diff
@@ -40,9 +54,26 @@ export function AlertingFlyout(props: Props) {
         alertTypeId: alertType,
         canChangeTrigger: false,
         initialValues,
+        metadata: {
+          environment,
+          serviceName,
+          transactionType,
+          start,
+          end,
+        } as AlertMetadata,
```

Comment:
Why did you manually set the type here? Should TS infers it automatically?

Reply:
No, `getAddAlertFlyout` doesn't allow specifying a generic for `metadata` and it's `Record<string, any>` by default: https://github.com/elastic/kibana/blob/c656608d85d15b684329ef1c68384c4b98d7606e/x-pack/plugins/triggers_actions_ui/public/types.ts#L285-L297

This `as` isn't needed to make the type check pass but just to ensure we're passing the values we ourselves expect.